### PR TITLE
Fix: Navigating using left/right arrows in the column layout no longer crashes the app

### LINKED ISSUE #10109: Bug: Using left/right arrow key in column layout crashes app
### Description

Using the left/right arrow key in the column layout to switch between columns crashes the app.

### Steps To Reproduce

1. Switch to the column layout
2. Select a folder
3. Use left/right arrow key to switch between the existing and newly opened column
4. See crash

### Steps to fix

- Fix crash that occurs when using keyboard navigation

### Files Version

v2.3.39

### Windows Version

Windows 11

### Relevant Assets/Logs

[debug.log](https://github.com/files-community/Files/files/9694991/debug.log)

### PR DESCRIPTION
**Pull request by QuaintMako**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #10109 

**What has been done**
- Programmatically handling the change of focus between the columns.
- Minor refactoring.

**Validation**
How did you test these changes?
- [X] Built and ran the app

**Screenshots**

CrashTabFixDraft

### CODE REVIEW

**Comment by QuaintMako** [comment]
The work for this PR is not complete yet, but I have an hard time finishing it.

There are several limitations currently preventing it to be considered fully functional:

1. When moving from the left or right, it will always refocus the first element of the `FileList`. When returning to the preview folder, it should focus the parent folder.
2. When moving the focus, the item is not "fully focused", as you can see on the gif. The user needs to press the down arrow to fully select the item.

If anyone fancy giving it a shot, I'll gladly take the help! In any case, I'm still looking into it.

**Comment by QuaintMako** [comment]
The work for this PR is almost done, but I cannot find a cleaner way to refresh the selection when backing up to the previous folder between line.222 and line.226 in `ColumnViewBrowser.xaml.cs`. If anyone has an idea, I'll gladly take it.

**Comment by QuaintMako** [comment]
> The work for this PR is almost done, but I cannot find a cleaner way to refresh the selection when backing up to the previous folder between line.222 and line.226 in `ColumnViewBrowser.xaml.cs`. If anyone has an idea, I'll gladly take it.

Cannot find a cleaner way to solve this problem, so I'm opening it for review.

File: src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs [hunk 1]
```diff
@@ -399,20 +399,24 @@ private async void FileList_PreviewKeyDown(object sender, KeyRoutedEventArgs e)
...
-					if ((ParentShellPageInstance as ColumnShellPage)?.ColumnParams.Column > 0)
-						FocusManager.TryMoveFocus(FocusNavigationDirection.Previous);
-					e.Handled = true;
-				}
+				if (IsRenamingItem || (ParentShellPageInstance is not null && ParentShellPageInstance.ToolbarViewModel.IsEditModeEnabled) )
+					return;
+
+				var currentBladeIndex = (ParentShellPageInstance is ColumnShellPage associatedColumnShellPage) ? associatedColumnShellPage.ColumnParams.Column : 0;
+				if (currentBladeIndex == 0)
+					return;
```
Comment (ferrariofilippo) [inline src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs:407]: I am not sure about two things:
- Is it worth double-checking `currentBladeIndex`? (Its value is checked both here and in `MoveFocusToPreviousBlade`)
- I think `e.Handled` should be set to true anyway, as you did for `VirtualKey.Right`

File: src/Files.App/Views/LayoutModes/ColumnViewBrowser.xaml.cs [hunk 2]
```diff
@@ -205,6 +205,52 @@ public void NavigateUp()
...
+				return;
+
+			DismissOtherBlades(currentBladeIndex);
+
+			var activeBlade = ColumnHost.ActiveBlades[currentBladeIndex - 1];
+			activeBlade.Focus(FocusState.Programmatic);
+
+			var activeBladeColumnViewBase = RetrieveBladeColumnViewBase(activeBlade);
+			if (activeBladeColumnViewBase == null)
+				return;
```
Comment (ferrariofilippo) [inline src/Files.App/Views/LayoutModes/ColumnViewBrowser.xaml.cs:220]: Can you change `==` to `is`? We have just changed this throughout the whole code base (#10326)

File: src/Files.App/Views/LayoutModes/ColumnViewBrowser.xaml.cs [hunk 3]
```diff
@@ -205,6 +205,52 @@ public void NavigateUp()
...
+		public void MoveFocusToNextBlade(int currentBladeIndex)
+		{
+			if (currentBladeIndex >= ColumnHost.ActiveBlades.Count)
+				return;
+
+			var activeBlade = ColumnHost.ActiveBlades[currentBladeIndex];
+			activeBlade.Focus(FocusState.Programmatic);
+
+			var activeBladeColumnViewBase = RetrieveBladeColumnViewBase(activeBlade);
+			if (activeBladeColumnViewBase != null)
```
Comment (ferrariofilippo) [inline src/Files.App/Views/LayoutModes/ColumnViewBrowser.xaml.cs:237]: Same as above

File: src/Files.App/Views/LayoutModes/ColumnViewBrowser.xaml.cs [hunk 4]
```diff
@@ -205,6 +205,52 @@ public void NavigateUp()
...
+				activeBladeColumnViewBase.FileList.SelectedIndex = 0;
+		}
+
+		private ColumnViewBase? RetrieveBladeColumnViewBase(BladeItem blade)
+		{
+			var activeBladeFrame = blade.Content as Frame;
+			if (activeBladeFrame == null)
```
Comment (ferrariofilippo) [inline src/Files.App/Views/LayoutModes/ColumnViewBrowser.xaml.cs:244]: Same as above

File: src/Files.App/Views/LayoutModes/ColumnViewBrowser.xaml.cs [hunk 5]
```diff
@@ -205,6 +205,52 @@ public void NavigateUp()
...
+				return null;
+
+			var activeBladePage = activeBladeFrame.Content as ColumnShellPage;
+			if (activeBladePage == null)
```
Comment (ferrariofilippo) [inline src/Files.App/Views/LayoutModes/ColumnViewBrowser.xaml.cs:248]: Same as above

**Review by ferrariofilippo — CHANGES_REQUESTED** [review]
There are some problems with the code style and lines 402 -> 410 (`src.Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs`), otherwise it is fine.

**Comment by ferrariofilippo** [comment]
I may have found a bug but I'm not sure if it's related to this PR and the way we focus a blade.
Sometimes, while navigating between columns, it happens that the focus bounces between the last two blades.
I post here a video to show what happens.

https://user-images.githubusercontent.com/102259289/201199645-6e1243e5-e200-4a0f-b79c-511233a5e7e7.mp4

**Comment by QuaintMako** [comment]
> I may have found a bug but I'm not sure if it's related to this PR and the way we focus a blade. Sometimes, while navigating between columns, it happens that the focus bounces between the last two blades. I post here a video to show what happens.

I'm going to investigate this. It is probably linked to this PR, since beforehand we would not even move between the blades.

**Comment by QuaintMako** [comment]
After messing around a bit with the feature, I think it's not linked to the PR at all. Moving up and down with the arrows messes the display as well. I did not touch those. I propose to open another issue about this<
Thoughts @ferrariofilippo ?

**Comment by ferrariofilippo** [comment]
> I propose to open another issue about this

I'll do so.

**Review by ferrariofilippo — APPROVED** [review]
LGTM

**Review by yaichenbaum — APPROVED** [review]
LGTM

**Pull request merged**